File: kirun-js/src/engine/runtime/expression/operators/unary/ArithmeticUnaryMinusOperator.ts
import { PrimitiveUtil } from '../../../../util/primitive/PrimitiveUtil';
import { Operation } from '../../Operation';
import { UnaryOperator } from './UnaryOperator';

export class ArithmeticUnaryMinusOperator extends UnaryOperator {
    public apply(t: any): any {
        this.nullCheck(t, Operation.UNARY_MINUS);

        PrimitiveUtil.findPrimitiveNumberType(t);

        return -t;
    }
}

File: kirun-py/src/kirun_py/runtime/expression/operators/unary/arithmetic_unary_minus_operator.py
from __future__ import annotations

from typing import Any

from kirun_py.util.primitive.primitive_util import PrimitiveUtil
from kirun_py.runtime.expression.operators.unary.unary_operator import UnaryOperator


class ArithmeticUnaryMinusOperator(UnaryOperator):

    def apply(self, t: Any) -> Any:
        from kirun_py.runtime.expression.operation import Operation
        self.null_check(t, Operation.UNARY_MINUS)
        PrimitiveUtil.find_primitive_number_type(t)
        return -t
